conditions/v1/conditions_test.go

```diff
@@ -113,20 +117,57 @@ func TestSetStatusCondition(t *testing.T) {
 					Message: "Degraded condition true",
 				},
 			},
+			expectChanged: true,
+		},
+		{
+			name: "no change",
+			testCondition: Condition{
+				Type:    ConditionDegraded,
+				Status:  "True",
+				Reason:  "TestingDegradedTrue",
+				Message: "Degraded condition true",
+			},
+			startConditions: &[]Condition{
+				{
+					Type:    ConditionDegraded,
+					Status:  "True",
+					Reason:  "TestingDegradedTrue",
+					Message: "Degraded condition true",
+				},
+			},
+			expectedConditions: &[]Condition{
+				{
+					Type:    ConditionDegraded,
+					Status:  "True",
+					Reason:  "TestingDegradedTrue",
+					Message: "Degraded condition true",
+				},
+			},
+			expectChanged: false,
 		},
 	}
 
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
-			SetStatusCondition(tc.startConditions, tc.testCondition)
-			compareConditions(t, tc.startConditions, tc.expectedConditions)
+			// Copy tc.startConditions so it doesn't get updated in place
+			startConditions := make([]Condition, len(*tc.startConditions))
+			copy(startConditions, *tc.startConditions)
```

Comment:
note latent bug in the test driver: previously the first `SetStatusCondition` was overwriting `tc.startConditions` so it was already in the expected state when it got to `SetStatusConditionNoHeartbeat`.